retranslate/rewrite-pos/rewrite-pos.js

```diff
@@ -0,0 +1,74 @@
+
+import gettextParser from "gettext-parser";
+import * as fs from "fs";
+import * as Path from 'path';
+
+const newSourcePath = process.env.NEW_SOURCE_PO || './public/i18n/de.txt';
+const inputFiles = process.env.PO_FILES || './public/i18n/';
+const extensions = (process.env.PO_EXTENSIONS || '.po,.txt').split(',');
+
+const newSourceContent = fs.readFileSync(newSourcePath);
+const newSourcePo = gettextParser.po.parse(newSourceContent);
+
+// takes two parsed po files & rewrites the ids of the first one based on the new sources translations
+function replaceMsgidWithMsgstr(parsedPo, newSourcePo) {
+  for (const contextKey in parsedPo.translations) {
+      // handle invalid contexts
+    if (!newSourcePo.translations[contextKey]) {
+      console.error(`new source translation missing context '${contextKey}'`);
+      continue;
+    }
+    const rewrittenContext = {};
+    for (const key in parsedPo.translations[contextKey]) {
+      // handle invalid strings
+      if (!newSourcePo.translations[contextKey][key]) {
+        console.error(`new source translation missing string '${key}'`);
+        continue;
+      }
+
+      const rewriteValue = newSourcePo.translations[contextKey][key];
+      const newValue = Object.assign({}, parsedPo.translations[contextKey][key]);
+      // translate base
+      if (rewriteValue.msgstr[0]) {
+        const newKey = rewriteValue.msgstr[0];
+        newValue.msgid = newKey;
+        rewrittenContext[newKey] = newValue;
+      } else {
+        console.warn(`new source translation missing for '${newValue.msgid}'`);
+        rewrittenContext[key] = newValue;
+      }
+
+      // retranslate plurals
+      if (newValue.msgid_plural) {
+        if (rewriteValue.msgstr[1]) {
+          const newKey = rewriteValue.msgstr[1];
+          newValue.msgid_plural = newKey;
```

Comment:
Does this work with languages that have more than two plural forms, e.g. Russian?

Reply:
Only the first plural form is used as the new plural id, this should not matter for the other plural forms